src/catchup/ApplyCheckpointWork.cpp

```diff
@@ -258,6 +268,32 @@ ApplyCheckpointWork::applyHistoryOfSingleLedger()
 BasicWork::State
 ApplyCheckpointWork::onRun()
 {
+    auto& hm = mApp.getHistoryManager();
+    if (mPublishTimer)
+    {
+        if (hm.publishQueueLength() > 0)
```

Comment:
you should use constants instead of `0` in both locations, and take the opportunity to define the two thresholds (0 is too aggressive, I think):
* increase the size of the queue needed to trigger a wait - let's say 32
* define the size of the queue to unblock applying ledgers - let's say 16 for now

Reply:
Seems like the way ConditionalWork is implemented, it'd require changing state inside the lambda to know whether to wait or apply when publish queue size it in between 16 and 32 (which I think is quite error-prone and also doesn't comply with ConditionFn contract). I added the lower boundary of 16 for now to preserve the const-ness of lambda, what do you think?

Reply:
What do you mean by changing state inside the lambda @marta-lokhova ? You mean the conditional is stateful? I don't think this is a problem: we do this all the time when we capture a pointer to a class.